openmicroscopy/openmicroscopy
# Remove pyramid

# What this PR does

Offer a CLI option to remove pyramid of a given endianness
This should allow to remove the corrupted pyramids using ``removepyramids``
This follow the approach taken by ``cleanse`` or ``fixpyramids`` (we could look at some unification at some stage/or move to a plugin)

# Testing this PR
Several options are possible (Indicated with ``--dry-run`` so that the pyramids are not deleted)
1.
``
bin/omero admin removepyramids --dry-run
``
will list the pyramids with little-endian=false to be deleted

2. 
``
bin/omero admin removepyramids  --dry-run --endian=little
``
will list the pyramids with little-endian=true to be deleted

3. 
``
bin/omero admin removepyramids --dry-run --endian=little --imported-after 2018/02/10
``
will list the pyramids with little-endian=true and imported after a given date to be deleted
Check is a valid date and a date in the future.
Nothing should be printed out for a date in the future

one image generating an OMERO pyramid have been imported on 15/02 for root:

*  in ``fake`` with little-endian=true

* Delete the pyramids with little-endian=true. The thumbnail on disk will also be deleted
* Note the values returned e.g. ``255``. ``255`` is the image ID
* Refresh the dataset
* The "clock" should be displayed instead of the thumbnail.
* When the pyramid creation is done, refresh again, a valid thumbnail should be created. 
The pyramid and thumbnails have been regenerated

# Related reading

https://trello.com/c/o3Ro0GXT/24-omero-import-large-tiffs-issue

Note:
This was initially added to ``admin`` since it required local access but the implementation has changed since.
It is kept in admin so it is in synch with the other methods e.g. ``fixpyramids``
